ddprof-lib/src/main/cpp/flightRecorder.cpp

```diff
@@ -752,11 +753,12 @@ void Recording::writeDatadogProfilerConfig(Buffer* buf,
                                 long allocInterval,
                                 long memleakInterval,
                                 long memleakCapacity,
+                                double memleakRatio,
                                 bool gcGenerations,
                                 int modeMask,
                                 const char* cpuEngine) {
     flushIfNeeded(buf, RECORDING_BUFFER_LIMIT
-    - (1 + 5 * MAX_VAR64_LENGTH + MAX_VAR32_LENGTH + 3 * MAX_STRING_LENGTH));
+    - (1 + 6 * MAX_VAR64_LENGTH + MAX_VAR32_LENGTH + 3 * MAX_STRING_LENGTH));
```

Comment:
This is just a safety margin. It should be ok to have `(2 + 5 * MAX_VAR64_LENGTH + MAX_VAR32_LENGTH + 3 * MAX_STRING_LENGTH))` because we can fit the percent number into one byte.

Reply:
Also, I tried recording it as a JFR float but the resulting recording was unreadable. I really didn't want to spend more time on that so I opted for recording the setting as a percentual value.